learning/tour-of-beam/frontend/lib/repositories/client/cloud_functions_client.dart

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+
+import '../../models/content_tree.dart';
+import '../models/get_sdks_response.dart';
+import 'client.dart';
+
+class CloudFunctionsTobClient extends TobClient {
+  @override
+  Future<GetSdksResponse> getSdks() async {
+    final json = await http.get(
+      Uri.parse(
+        'https://us-central1-tour-of-beam-2.cloudfunctions.net/getSdkList',
```

Comment:
May we consider not hard-coding URLs in the codebase?

Reply:
Sure. It's just a temporary URL of a test backend.

In production, we will have a deployment workflow, and the production backend URL will be emitted at some point during that workflow. When it is ready, we will decide how to capture it for the frontend. It may be a generated `.dart` file with constants (if the URL is more or less permanent) or a text file read at startup (if the URL may change with backend re-deployments).

Without knowing the future infrastructure details, the best we can do is to extract the domain `us-central1-tour-of-beam-2.cloudfunctions.net` to some constant to not repeat it for two functions which is a too minor improvement. I suggest we do this in the next ToB PR which is already based on this unmerged one.

Reply:
In this PR, could we see how a configuration would solve preventing this hard coded value?

Reply:
I made `config.dart` as a temporary solution.
It now has:

1. Cloud functions URL which is a function of the project region and the project ID. These both will be known when deploying cloud functions. When that workflow is ready, we will generate a config with those constants.

2. Playground backend URLs. These are permanent but they are also known when deploying the backend. When Playground backend deployment workflow is fixed, we will be generating both Playground's config an ToB's config.